client/gutenboarding/store/actions.ts

```diff
@@ -0,0 +1,17 @@
+/**
+ * External dependencies
+ */
+import { Action, ActionCreator } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import ActionTypes from './action-types';
+
+export interface SetSiteTypeAction extends Action< ActionTypes.SET_SITE_TYPE > {
+	payload: string;
+}
+export const setSiteType: ActionCreator< SetSiteTypeAction > = ( siteType: string ) => ( {
+	type: ActionTypes.SET_SITE_TYPE,
+	payload: siteType,
```

Comment:
How about using a less generic name for `payload`? Just `siteType` maybe?

```suggestion
	siteType,
```